packages/firestore/src/api/database.ts

```diff
@@ -612,37 +618,14 @@ export class Firestore implements firestore.FirebaseFirestore, FirebaseService {
     return new WriteBatch(this);
   }
 
-  static get logLevel(): firestore.LogLevel {
-    switch (getLogLevel()) {
-      case LogLevel.DEBUG:
-        return 'debug';
-      case LogLevel.SILENT:
-        return 'silent';
-      default:
-        // The default log level is error
-        return 'error';
-    }
+  static get logLevel(): LogLevel {
+    return getLogLevel();
   }
 
-  static setLogLevel(level: firestore.LogLevel): void {
+  static setLogLevel(level: LogLevelString): void {
     validateExactNumberOfArgs('Firestore.setLogLevel', arguments, 1);
     validateArgType('Firestore.setLogLevel', 'non-empty string', 1, level);
```

Comment:
Instead of this, you may want to call https://osscs.corp.google.com/firebase-sdk/firebase-js-sdk/+/master:packages/firestore/src/util/input_validation.ts;l=303?q=validate%20enum&ss=firebase-sdk%2Ffirebase-js-sdk

If you don't validate the enum constant here, `setLogLevel` sets the log level to undefined.